src/SmoFacade/BackupFileTools.cs

```diff
@@ -17,7 +17,9 @@ public enum BackupType
 
     public static bool IsUrl(string path)
     {
-      return Regex.IsMatch(path, @"(http(|s):\/)(\/[^\s]+)+\.([a-zA-Z]+)$");
+      return Uri.TryCreate(path, UriKind.Absolute, out var uriResult)
+             && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps || uriResult.IsUnc)
+             && !(path.EndsWith(@"/") || path.EndsWith(@"\"));

```

Comment:
This function does much more than the signature `IsUrl` would indicate.
I'd rename to `IsValidUriFile`

Also, the logic `&& !(path.EndsWith(@"/") || path.EndsWith(@"\"))` can be improved upon.  
You omitted the test below for the case `http://storage-account.blob.core.windows.net/container` which doesn't have a trailing slash.
That case would fail this test here.
I think using a regex or [Ssytem.IO.Path library](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.hasextension?view=net-5.0) to validate the path has an extension here would be more ideal

Reply:
Sure will change the name to something more reflective of the implementation.

The issue I found here is that not all files have an extension. That is why I also changed the test cases to reflect this.
Although I think I just had one positive test case there that showed this and that too only for UNC (`[InlineData(@"\\server\file")]`) - so can add some more there?

If we do want to restrict to just files with extensions I think we can do something like:
`new Uri(url).AbsolutePath.Split('/').Last().Contains('.');` or maybe just check for `Path.GetExtension(url)`
https://stackoverflow.com/a/53262979/15521092

A more concrete check would be to see if the file actually exists - `File.Exists(url)`
But in some cases the consumer of this module might not have access to the files themselves (I am facing this now) - so I think we should stick to just checking the syntax of the url.

Please let me know your thoughts.

Reply:
What files don't have extensions?  Surely all of our SQL backup files do.  Consult with Angelo and let me know if that assumption is wrong for some reason.

`System.IO.Path.HasExtension` is all that should be needed.  https://docs.microsoft.com/en-us/dotnet/api/system.io.path.hasextension?view=net-5.0

`File.Exists()` isn't going to work for URL file paths

Reply:
Ah you are right. I was thinking of files in general (filed don't need extensions to work - [ref](https://superuser.com/questions/731323/does-one-really-need-to-use-file-extensions-for-files-stored-on-their-hard-drive#:~:text=Short%20answer%3A%20No%2C%20we%20don,then%20double%20click%20on%20it.))

But maybe for SQL backup files we will always have an extension like `.trn`/`.full`/`.diff` etc..
Will double check and get back

Reply:
It is safe to assume that all backup files will have extensions.  This will definitely be true for any backups created by the standard SQL backup jobs, and that's all we have to care about.  Even manual backups typically are created with an extension, though not necessarily the same as those created by our jobs (.full, .diff, .trn).  Yes it is technically possible for someone to come in and take a manual backup that does not have an extension, but we can say those backups won't be supported.

Reply:
Noting I haven't re-reviewed because this hasn't been changed yet

Reply:
Done!